app/lib/features/chat/widgets/emoji_reaction_item.dart

```diff
@@ -15,44 +16,49 @@ class EmojiReactionItem extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final profile = ref.watch(memberProfileProvider(userId));
-
-    return ListTile(
-      leading: profile.when(
-        data: (data) => ActerAvatar(
-          mode: DisplayMode.User,
-          uniqueId: userId,
-          size: data.hasAvatar() ? 18 : 36,
-          avatar: data.getAvatarImage(),
-          displayName: data.displayName,
-        ),
-        loading: () => const Text('loading'),
-        error: (e, t) {
-          debugPrint('loading avatar failed: $e');
-          return ActerAvatar(
-            uniqueId: userId,
-            displayName: userId,
-            mode: DisplayMode.User,
-            size: 36,
-          );
-        },
-      ),
-      title: profile.when(
-        data: (data) => Text(data.displayName ?? userId),
-        loading: () => Text(userId),
-        error: (e, s) => Text('loading profile failed: $e'),
-      ),
-      subtitle: Text(userId),
-      trailing: Wrap(
-        children: emojis
-            .map(
-              (emoji) => Text(
-                emoji,
-                style: EmojiConfig.emojiTextStyle,
-              ),
-            )
-            .toList(),
-      ),
+    final member = ref.watch(chatMemberProvider(userId));
+    return member.maybeWhen(
+      data: (member) {
+        final profile = ref.watch(memberProfileProvider(member));
```

Comment:
do I read these correctly: by splitting the provider into two, we now have a bunch of `.maybeWhen` that don't really equate to anything as the widgets aren't drawn unless there is a valid _userId_ .

May I suggest adding a provider that just encapsulates these two providers as one again? Call it `memberProfileFromUserIdProvider` and then we can still just have a single provider and thus a single level of indentation as the other is not needed either way ...

Reply:
Correct, I think we should merge these into one. I would like to keep ```memberProfileProvider``` as it is because its re-usable, needs ```Member``` object which doesn't need additional reference of being part of ```Convo``` or ```Space```.

Reply:
Correct, we should merge these into one. But that leads to another question. How would we able to reference ```userId``` to ```Convo``` or ```Space```? I would still like to keep ```memberProfileProvider``` as taking ```Member``` since it doesn't need prior information of what it's a part of and thus being reusable. ```userId``` is dependent on additional context of which hierarchy it belongs.

Reply:
Not saying we should merge them. They can exist as they are, but we can _add_ a provider (on top) that combines both providers. Everything else can stay the same.